source.lua

```diff
@@ -127,18 +181,52 @@ function main(args)
         if not (liquidArg == 'magma' or liquidArg == 'water') then
             qerror('Liquid must be either "water" or "magma"')
         end
-        if not IsFlowPassable(targetPos) then
+        if not is_flow_passable(targetPos) then
             qerror("Tile not flow passable: I'm afraid I can't let you do that, Dave.")
         end
         local amountArg = tonumber(args[3]) or 7
-        AddLiquidSource(targetPos, liquidArg, amountArg)
+        add_liquid_source(targetPos, liquidArg, amountArg)
         print(('Added %s %d at %s'):format(liquidArg, amountArg, formatPos(targetPos)))
         return
     end
+end
 
-    print(dfhack.script_help())
+dfhack.onStateChange[GLOBAL_KEY] = function(sc)
+    if sc ~= SC_MAP_LOADED or df.global.gamemode ~= df.game_mode.DWARF then
+        return
+    end
+    local persisted_data = json.decode(persist.GlobalTable[GLOBAL_KEY] or '') or {}
+    -- sometimes the keys come back as strings; fix that up
+    for k,v in pairs(persisted_data) do
+        if type(k) == 'string' then
+            persisted_data[tonumber(k)] = v
+            persisted_data[k] = nil
+        end
+    end
+    g_sources_list = persisted_data
+    load_liquid_source(g_sources_list)
+end
+
+if dfhack.internal.IN_TEST then
+    unit_test_hooks = {
+        clear_watched_job_matchers=clear_watched_job_matchers,
+        on_new_job=on_new_job,
+        status=status,
+        boost=boost,
+        boost_and_watch=boost_and_watch,
+        remove_watch=remove_watch,
+        print_current_jobs=print_current_jobs,
+        print_registry=print_registry,
+        parse_commandline=parse_commandline,
+    }
+end
+
+if dfhack_flags.module then
+    return
 end
 
 if not dfhack_flags.module then
     main({...})
 end
+
+persist_state(g_sources_list)
```

Comment:
this should be inside the if block with `main` since its purpose is to persist any state changes that `main` made